pkg/arvo/app/chat-cli.hoon

```diff
@@ -710,177 +674,68 @@
-      !>   ^-  action:group-store
-      ?:  =(u.whitelist allow)
-        [%add-members rid ships]
-      [%remove-members rid ships]
-    ::  +join: sync with remote mailbox
-    ::
-    ++  join
-      |=  [=target gyf=(unit char) ask-history=(unit ?)]
-      ^-  (quip card _state)
-      =^  moz  state
-        ?.  ?=(^ gyf)  [~ state]
-        (bind-glyph u.gyf target)
-      =.  audience  [target ~ ~]
-      =;  =card
-        [[card prompt:sh-out moz] state]
-      ::TODO  ideally we'd check permission first. attempting this and failing
-      ::      gives ugly %chat-hook-reap
-      %^  act  %do-join  %chat-view
-      :-  %chat-view-action
-      !>  ^-  action:view
-      [%join ship.target (target-to-path target) (fall ask-history %.y)]
-    ::  +leave: unsync & destroy mailbox
+      ?~  target
+        [[(show-chats:sh-out ~(tap in viewing))]~ state]
+      ::  only view existing chat-type graphs
+      ::
+      ?.  (is-chat-graph target)
+        [[(note:sh-out "no such chat")]~ state]
+      =^  cards  state
+        ?:  (~(has by bound) target)
+          [~ state]
+        (bind-default-glyph target)
+      :-  cards
+      state(viewing (~(put in viewing) target))
+    ::  +flee: stop printing messages from a resource
     ::
-    ::TODO  allow us to "mute" local chats using this
-    ++  leave
+    ++  flee
       |=  =target
-      =-  [[- ~] state]
-      ?:  =(our-self ship.target)
-        %-  print:sh-out
-        "can't ;leave local chats, maybe use ;delete instead"
-      %^  act  %do-leave  %chat-hook
-      :-  %chat-hook-action
-      !>  ^-  action:hook
-      [%remove (target-to-path target)]
+      ^-  (quip card _state)
+      [~ state(viewing (~(del in viewing) target))]
     ::  +say: send messages
     ::
     ++  say
-      |=  =letter:store
+      |=  msg=content:post
       ^-  (quip card _state)
-      ~!  bowl
       =/  =serial:store  (shaf %msg-uid eny.bowl)
       :_  state
-      ^-  (list card)
-      %+  turn  ~(tap in audience)
-      |=  =target
-      %^  act  %out-message  %chat-hook
-      :-  %chat-action
-      !>  ^-  action:store
-      :+  %message  (target-to-path target)
-      [serial *@ our-self now.bowl letter]
+      :_  ~
+      ::TODO  move creation into lib?
+      %^  act  %out-message
+        %graph-push-hook
+      :-  %graph-update
+      !>  ^-  update:graph
+      :+  %0  now.bowl
+      :+  %add-nodes  audience
+      %-  ~(put by *(map index:post node:graph))
+      :-  ~[now.bowl]
+      :_  *internal-graph:graph
+      ^-  post:post
+      [our-self ~[now.bowl] now.bowl [msg]~ ~ ~]
     ::  +eval: run hoon, send code and result as message
     ::
     ::    this double-virtualizes and clams to disable .^ for security reasons
     ::
     ++  eval
       |=  [txt=cord exe=hoon]
-      (say %code txt (eval:store bowl exe))
+      ~&  %eval-tmp-disabled
+      [~ state]
+      ::TODO  why -find.eval??
+      :: (say %code txt (eval:store bowl exe))
```

Comment:
idk what's up with this, by the way. Nothing's shadowing `store`, nothing about this even changed, but it just won't resolve to the right thing. What dumb mistake could I be overlooking?